api/pkg/apis/v1alpha1/managers/catalogs/catalogs-manager.go

```diff
@@ -234,9 +236,9 @@ func (t *CatalogsManager) ListSpec(ctx context.Context) ([]model.CatalogState, e
 	}
 	return ret, nil
 }
-func (g *CatalogsManager) setProviderDataIfNecessary(ctx context.Context) error {
+func (g *CatalogsManager) setProviderDataIfNecessary(ctx context.Context, scope string) error {
 	if !g.GraphProvider.IsPure() {
-		catalogs, err := g.ListSpec(ctx)
+		catalogs, err := g.ListSpec(ctx, scope)
```

Comment:
should set graph data be scope-wide? 
Even if graph query is scope-wide, graph data can also have all catalog